src/service_broker/service_broker/nextflow/scripts/local_nextflow.nf

```diff
@@ -46,21 +50,44 @@ process ocrd_cis_ocropy_binarize {
   echo true
 
   input:
-    path mets_file 
+    path mets_file
+    path dir_name
+  output:
     path dir_name
-  
+
   script:
   """
   docker run --rm -v ${params.volume_dir}:/data -w /data -- ocrd/all:maximum ocrd-cis-ocropy-binarize -m ${mets_file} -I ${dir_name} -O "OCR-D-BIN"
   """
 }
 
+process save_to_olahd {
+  // limit instances to 1
+  // (force to run sequentially)
+  maxForks 1
+  echo true
+
+  input:
+    path mets_file
+    // variable only needed to cause waiting for the completition of other processors
+    path dir_name
+    val olahd_endpoint
+    val olahd_username
+    val olahd_password
+
+  script:
+  """
+  docker run --rm -v ${params.volume_dir}:/data -w /data -- ocrd/all:maximum ocrd-olahd-client -m ${mets_file} -p '{"endpoint": "${olahd_endpoint}", "username": "${olahd_username}", "password": "${olahd_password}"}'
+  """
+}
+
 // This is the main workflow
 workflow {
 
   main:
     download_workspace(params.mets, params.file_group)
     input_dir_ch = Channel.fromPath(params.reads, type: 'dir')
     ocrd_cis_ocropy_binarize(params.mets, input_dir_ch)
-    
+    if (params.olahd_username)
+        save_to_olahd(params.mets, ocrd_cis_ocropy_binarize.out, params.olahd_endpoint, params.olahd_username, params.olahd_password)
```

Comment:
Here, we should check if all olahd parameters were set. Not just the username. 

Also, we should consider parameter validation as a first step of the workflow. Imagine a scenario where we run a workflow for some time and then the workflow fails in the last processor because not all required parameters were set.

Reply:
I added a check for all 3 parameter. Validation of parameter beforehand is still missing. I have try to think about how to do that. But it should be possible to just try to authenticate to olahd.